EIPS/eip-rpc.md

```diff
@@ -0,0 +1,2314 @@
+}
+```
+---
+</details>
+
+<details>
+<summary><code><strong>eth_getBlockTransactionCountByNumber</strong></code></summary>
+
+#### Description
+
+Returns the number of transactions in a block specified by block number
+
+#### Parameters
+
+|#|Type|Description|
+|-|-|-|
+|1|{[`Quantity`](#quantity)\|`string`}|block number, or one of `"latest"`, `"earliest"` or `"pending"`|
+
+#### Returns
+
+{[`Quantity`](#quantity)} - number of transactions in the specified block
+
+#### Example
+
+```sh
+# Request
+curl -X POST --data '{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "method": "eth_getBlockTransactionCountByNumber",
+    "params": ["0xe8"]
+}' <url>
+
+# Response
+{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "result": "0xa"
+}
+```
+---
+</details>
+
+<details>
+<summary><code><strong>eth_getCode</strong></code></summary>
+
+#### Description
+
+Returns the contract code stored at a given address
+
+#### Parameters
+
+|#|Type|Description|
+|-|-|-|
+|1|{[`Data`](#data)}|address to query for code|
+|2|{[`Quantity`](#quantity)\|`string`}|block number, or one of `"latest"`, `"earliest"` or `"pending"`|
+
+#### Returns
+
+{[`Data`](#data)} - code from the specified address
+
+#### Example
+
+```sh
+# Request
+curl -X POST --data '{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "method": "eth_getCode",
+    "params": ["0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b", "0x2"]
+}' <url>
+
+# Response
+{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "result": "0x600160008035811a818181146012578301005b601b6001356025565b8060005260206000f25b600060078202905091905056"
+}
+```
+---
+</details>
+
+<details>
+<summary><code><strong>eth_getFilterChanges</strong></code></summary>
+
+#### Description
+
+Returns a list of all logs based on filter ID since the last log retrieval
+
+#### Parameters
+
+|#|Type|Description|
+|-|-|-|
+|1|{[`Quantity`](#quantity)}|ID of the filter|
+
+#### Returns
+
+{`Array<Log>`} - array of log objects with the following members:
+
+- {[`Data`](#data)} `address` - address from which this log originated
+- {[`Data`](#data)} `blockHash` - hash of block containing this log or `null` if pending
+- {[`Data`](#data)} `data` - contains the non-indexed arguments of the log
+- {[`Data`](#data)} `transactionHash` - hash of the transaction that created this log or `null` if pending
+- {[`Quantity`](#quantity)} `blockNumber` - number of block containing this log or `null` if pending
+- {[`Quantity`](#quantity)} `logIndex` - index of this log within its block or `null` if pending
+- {[`Quantity`](#quantity)} `transactionIndex` - index of the transaction that created this log or `null` if pending
+- {[`Data[]`](#data)} `topics` - list of order-dependent topics
+- {`boolean`} `removed` - `true` if this filter has been destroyed and is invalid
+
+**Note:** The return value of `eth_getFilterChanges` when retrieving logs from `eth_newBlockFilter` and `eth_newPendingTransactionFilter` filters will be an array of hashes, not an array of Log objects.
```

Comment:
Since filters can timeout if not queried regularly, it seems appropriate to provide a *recommended* error code for clients to use when the provided `filder_id` is not found.

Reply:
Another good catch. I believe geth currently returns `-32000 - Invalid input` for this, but I feel that `-32001 - Resource not found` is more appropriate.